yroots/subdivision.py

```diff
@@ -99,61 +99,60 @@ def solve(funcs, a, b, rel_approx_tol=1.e-15, abs_approx_tol=1.e-12,
     -------
     zeros : numpy array
         The common zeros of the polynomials. Each row is a root.
-    '''
-    #Detect the dimension
+    """
+    # Detect the dimension
     if not isinstance(funcs,list):
         dim = 1
     else:
         dim = len(funcs)
 
-    #make a and b the right type
+    # make a and b the right type
     a = np.float64(a)
     b = np.float64(b)
 
     # Choose an appropriate max degree for the given dimension if none is specified.
     if deg is None:
-        deg_dim = {1: 50, 2:9, 3:5, 4:3}
+        deg_dim = {1: 100, 2:20, 3:9, 4:9}
         if dim > 4:
             deg = 2
         else:
             deg = deg_dim[dim]
 
     # Choose an appropriate target degree if none is specified
     if target_deg is None:
-        if dim != 2:
-            target_deg = deg
-        else:
+        if dim > 4:
             target_deg = 2
+        else:
+            target_deg = 3
 
-    #Sets up the tolerances.
+    # Sets up the tolerances.
     tols = Tolerances(rel_approx_tol=rel_approx_tol,
                       abs_approx_tol=abs_approx_tol,
                       max_cond_num=max_cond_num,
                       good_zeros_factor=good_zeros_factor,
                       min_good_zeros_tol=min_good_zeros_tol,
                       check_eval_error=check_eval_error,
-                      check_eval_freq=check_eval_freq)
+                      check_eval_freq=check_eval_freq,
+                      target_tol=target_tol)
     tols.nextTols()
 
-    #Set up the interval data and root tracker classes
+    # Set up the interval data and root tracker classes
     interval_data = IntervalData(a,b)
     root_tracker = RootTracker()
 
     if dim == 1:
         solve_func = subdivision_solve_1d
-        deg = 100
-        target_deg = 100 # Target degree doesn't mean anything in 1D.
```

Comment:
By taking this out I think we set target_deg for 1D to 3. Is that really what we want? We can do very high degree approximations successfully with 1D because you just have to construct the colleague matrix, and it doesn't matter if that is quite large.

Reply:
Yeah, I removed these lines to work with our dictionary of defaults more. You're right that this sets our 1D target_deg to 3, but the 1D case doesn't actually use target_deg at all. It just needs it as a parameter so that we can call solve_func for 1D and nD cases identically. To make that more explicit, I suppose we could reinclude these lines (or at least set the target_deg for 1D to be the same as deg).